packages/react-router/index.tsx

```diff
@@ -240,10 +241,12 @@ export function Router({
   );
 
   return (
-    <LocationContext.Provider
-      children={children}
-      value={{ action, location, navigator, static: staticProp }}
-    />
+    <NavigatorContext.Provider value={navigator}>
+      <LocationContext.Provider
+        children={children}
+        value={{ action, location, static: staticProp }}
```

Comment:
I just double checked and the only place the `static` value is ever used is in the `Navigate` component. Maybe it makes more sense to include it in the `NavigationContext`?

I really need to refresh my memory on how changes to context objects cause rerenders... I was under the impression that they don't, so it wouldn't mean we need to memoize the object. But now I'm wondering.

Reply:
I believe the optimal pattern is using `useMemo` for the `value` and having the `children` element be a single, memoized component. In this sandbox, if I use the non-memoized component or pass `{ color, shape }` instead of `value`, I get console logs in the component that consumes the context.

https://codesandbox.io/s/memoized-context-check-riqfs